tests/unittests/tests-fmt/tests-fmt.c

```diff
@@ -61,6 +108,52 @@ static void test_fmt_bytes_hex_reverse(void)
     TEST_ASSERT_EQUAL_STRING("06070809", (char *) out);
 }
 
+static void test_fmt_hex_bytes(void)
+{
+    uint8_t val;
+    uint8_t bytes = fmt_hex_bytes(&val, "");
+    TEST_ASSERT_EQUAL_INT(0, bytes);
+    TEST_ASSERT_EQUAL_INT(0, val);
+
+    bytes = fmt_hex_bytes(&val, "A");
+    TEST_ASSERT_EQUAL_INT(0, bytes);
+
+    char hex2[3] = "00";
+    uint8_t val1[1] = { 0 };
+    bytes = fmt_hex_bytes(val1, hex2);
+    TEST_ASSERT_EQUAL_INT(1, bytes);
+    TEST_ASSERT_EQUAL_INT(0, val1[0]);
+
+    memcpy(hex2, "AB", 2);
+    hex2[2] = '\0';
+    val1[0] = 0;
+    bytes = fmt_hex_bytes(val1, hex2);
+    TEST_ASSERT_EQUAL_INT(1, bytes);
+    TEST_ASSERT_EQUAL_INT(0xAB, val1[0]);
+
+    memcpy(hex2, "CD", 2);
+    hex2[2] = '\0';
+    val1[0] = 0;
+    bytes = fmt_hex_bytes(val1, hex2);
+    TEST_ASSERT_EQUAL_INT(1, bytes);
+    TEST_ASSERT_EQUAL_INT(0xCD, val1[0]);
+
+    memcpy(hex2, "EF", 2);
+    hex2[2] = '\0';
+    val1[0] = 0;
+    bytes = fmt_hex_bytes(val1, hex2);
+    TEST_ASSERT_EQUAL_INT(1, bytes);
+    TEST_ASSERT_EQUAL_INT(0xEF, val1[0]);
+
+    char hex4[6] = "0102aF";
+    uint8_t val3[3];
+    bytes = fmt_hex_bytes(val3, hex4);
+    TEST_ASSERT_EQUAL_INT(3, bytes);
+    TEST_ASSERT_EQUAL_INT(1, val3[0]);
+    TEST_ASSERT_EQUAL_INT(2, val3[1]);
+    TEST_ASSERT_EQUAL_INT(0xAF, val3[2]);
```

Comment:
What happens if a character in the string is not in `[0-9a-fA-F]`?

Reply:
it doesn't work. It's possible to update but I want to keep this simple and trust the user. I added a doxygen comment for this.